.github/workflows/multi-arch-build.yaml

```diff
@@ -0,0 +1,175 @@
+      CONTAINERS_QUAY_REGISTRY: quay.io/containers
+      # list of architectures for build
+      PLATFORMS: linux/amd64,linux/s390x,linux/ppc64le,linux/arm64
+
+    # build several images (upstream, testing, stable) in parallel
+    strategy:
+      matrix:
+        # Builds are located under contrib/podmanimage/<source> directory
+        source:
+          - upstream
+          - testing
+          - stable
+    runs-on: ubuntu-latest
+    # internal registry caches build for inspection before push
+    services:
+      registry:
+        image: quay.io/libpod/registry:2
+        ports:
+          - 5000:5000
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+          install: true
+
+      - name: Build and locally push Podman
+        uses: docker/build-push-action@v2
+        with:
+          context: contrib/podmanimage/${{ matrix.source }}
+          file: ./contrib/podmanimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: localhost:5000/podman/${{ matrix.source }}
+
+      # Simple verification that container works + grab version number
+      - name: amd64 container sniff test
+        id: sniff_test
+        run: |
+          VERSION_OUTPUT="$(docker run localhost:5000/podman/${{ matrix.source }} \
+                            podman --storage-driver=vfs version)"
+          echo "$VERSION_OUTPUT"
+          VERSION=$(grep -Em1 '^Version: ' <<<"$VERSION_OUTPUT" | awk '{print $2}')
+          test -n "$VERSION"
+          echo "::set-output name=version::${VERSION}"
+
+      # Generate image FQINs, labels, check whether to push
+      - name: Generate image information
+        id: image_info
+        run: |
+          if [[ "${{ matrix.source }}" == 'stable' ]]; then
+            # quay.io/podman/stable:vX.X.X
+            PUSH=$(skopeo list-tags \
+                     docker://${{ env.PODMAN_QUAY_REGISTRY }}/stable | \
+                   jq -r .Tags[] | \
+                   grep -q "^${VERSION}$" && \
+                   echo 'false' || echo 'true')
+            FQIN='${{ env.PODMAN_QUAY_REGISTRY }}/stable:v${{ steps.sniff_test.outputs.version }}' # workaround vim syntax-hilighting bug: '
+            # Only push if version tag does not exist
+            if [[ "$PUSH" == "true" ]]; then
+              echo "Will push $FQIN"
+              echo "::set-output name=podman_push::${PUSH}"
+              echo "::set-output name=podman_fqin::${FQIN}"
+            fi
+
+            # quay.io/contaners/podman:vX.X.X
+            unset PUSH
+            PUSH=$(skopeo list-tags \
+                       docker://${{env.CONTAINERS_QUAY_REGISTRY}}/podman | \
+                   jq -r .Tags[] | \
+                   grep -q "^${VERSION}$" && \
```

Comment:
Sure we could...though I *just* noticed this in a man page-

    [egrep and fgrep]...These variants are deprecated, but are provided
    for backward compatibility.

Otherwise my only other excuse is this is how it was done in the buildah version of this action :grin: 

I'll change it...it will be more clear your way.